flink-connectors/flink-connector-hbase/src/main/java/org/apache/flink/connector/hbase/source/AbstractTableInputFormat.java

```diff
@@ -186,24 +193,22 @@ public void close() throws IOException {
 			if (resultScanner != null) {
 				resultScanner.close();
 			}
-		} finally {
-			resultScanner = null;
-		}
-	}
-
-	@Override
-	public void closeInputFormat() throws IOException {
-		try {
 			if (table != null) {
 				table.close();
 			}
+			if (connection != null) {
+				connection.close();
+			}
```

Comment:
We will leak the connection if table.close() or resultScanner.close() fails with an exception.
I believe the proper way would be to wrap each close in a separate try block, and catch and log the error.

Reply:
@rmetzger You are right, but `resultScanner.close()` doesn't throw exception. Therefore, I just add the close of the connection into catch block.